rbd/mirror_group.go

```diff
@@ -0,0 +1,493 @@
+static inline int rbd_mirror_group_get_global_status_dlsym(void *fn, rados_ioctx_t p, const char *name,
+                                                           rbd_mirror_group_global_status_t *mirror_group_status,
+														   size_t status_size){
+  // cast function pointer fn to rbd_mirror_group_get_global_status and call the function
+  return ((rbd_mirror_group_get_global_status_fn) fn)(p, name, mirror_group_status, status_size);
+}
+*/
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"unsafe"
+
+	"github.com/ceph/go-ceph/internal/cutil"
+	"github.com/ceph/go-ceph/internal/dlsym"
+	"github.com/ceph/go-ceph/rados"
+)
+
+// MirrorGroupEnable will enable mirroring for a group using the specified mode.
+//
+// Implements:
+//
+//	int rbd_mirror_group_enable(rados_ioctx_t p, const char *name,
+//	  							rbd_mirror_image_mode_t mirror_image_mode,
+//									uint32_t flags);
+func MirrorGroupEnable(groupIoctx *rados.IOContext, groupName string, mode ImageMirrorMode) error {
+	cGroupName := C.CString(groupName)
+	defer C.free(unsafe.Pointer(cGroupName))
+
+	rbdMirrorGroupEnable, rbdMirrorGroupEnableErr := dlsym.LookupSymbol("rbd_mirror_group_enable")
+	if rbdMirrorGroupEnableErr != nil {
+		return fmt.Errorf("%w: %w", ErrNotImplemented, rbdMirrorGroupEnableErr)
+	}
+
+	ret := C.rbd_mirror_group_enable_dlsym(
+		rbdMirrorGroupEnable,
+		cephIoctx(groupIoctx),
+		cGroupName,
+		C.rbd_mirror_image_mode_t(mode),
+		(C.uint32_t)(2),
+	)
+
+	return getError(ret)
+}
+
+// MirrorGroupDisable will disable mirroring for a group
+//
+// Implements:
+//
+//	int rbd_mirror_group_disable(rados_ioctx_t p, const char *name,
+//	  							bool force)
+func MirrorGroupDisable(groupIoctx *rados.IOContext, groupName string, force bool) error {
+	cGroupName := C.CString(groupName)
+	defer C.free(unsafe.Pointer(cGroupName))
+
+	rbdMirrorGroupDisable, rbdMirrorGroupDisableErr := dlsym.LookupSymbol("rbd_mirror_group_disable")
+	if rbdMirrorGroupDisableErr != nil {
+		return fmt.Errorf("%w: %w", ErrNotImplemented, rbdMirrorGroupDisableErr)
+	}
+
+	ret := C.rbd_mirror_group_disable_dlsym(
+		rbdMirrorGroupDisable,
+		cephIoctx(groupIoctx),
+		cGroupName,
+		C.bool(force))
+
+	return getError(ret)
+}
+
+// MirrorGroupPromote will promote the mirrored group to primary state
+//
+// Implements:
+//
+//	int rbd_mirror_group_promote(rados_ioctx_t p, const char *name,
+//	  							uint32_t flags, bool force)
+func MirrorGroupPromote(groupIoctx *rados.IOContext, groupName string, force bool) error {
+	cGroupName := C.CString(groupName)
+	defer C.free(unsafe.Pointer(cGroupName))
+
+	rbdMirrorGroupPromote, rbdMirrorGroupPromoteErr := dlsym.LookupSymbol("rbd_mirror_group_promote")
+	if rbdMirrorGroupPromoteErr != nil {
+		return fmt.Errorf("%w: %w", ErrNotImplemented, rbdMirrorGroupPromoteErr)
+	}
+
+	ret := C.rbd_mirror_group_promote_dlsym(
+		rbdMirrorGroupPromote,
+		cephIoctx(groupIoctx),
+		cGroupName,
+		(C.uint32_t)(0),
```

Comment:
define a const for the flag and add documentation on what each value mean

Reply:
Done